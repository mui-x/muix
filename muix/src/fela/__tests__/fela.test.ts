import { dumpAtomized, getRenderer } from "reactxx-fela";
import { initPlatform } from "reactxx-tests";

test("fela", () => {
  initPlatform(true);
  const classNames = getRenderer().renderRuleEx(
    {
      color: "red",
      margin: 10,
      ":hover": {
        margin: 5,
        "@media (min-width: 768px)": {
          height: 100,
          ":active": {
            width: 700,
            color: "red"
          }
        }
      }
      // }
    },
    "root"
  );
  const res = dumpAtomized(classNames);
  expect(res).toMatchSnapshot();
});
