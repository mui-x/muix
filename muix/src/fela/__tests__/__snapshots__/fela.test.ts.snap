// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fela 1`] = `
Array [
  ".a { color:red /*root*/ }",
  ".b { margin:10px /*root*/ }",
  ".c:hover { margin:5px /*root*/ }",
  "@media (min-width: 768px).d:hover { height:100px /*root*/ }",
  "@media (min-width: 768px).e:hover:active { width:700px /*root*/ }",
  "@media (min-width: 768px).f:hover:active { color:red /*root*/ }",
]
`;
